src/ProjectTemplates/Web.Spa.ProjectTemplates/content/React-CSharp/ClientApp/src/App.test.js

```diff
@@ -3,10 +3,11 @@ import ReactDOM from 'react-dom';
 import { MemoryRouter } from 'react-router-dom';
 import App from './App';
 
-it('renders without crashing', () => {
+it('renders without crashing', async () => {
   const div = document.createElement('div');
   ReactDOM.render(
     <MemoryRouter>
       <App />
     </MemoryRouter>, div);
+  await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
Should this new line be here? It doesn't seem necessary for the test.

Reply:
It is, when we are doing auth the test becomes async and fails if not. (Throws indicating that the component has been unmounted). Technically a promise resolve method passed to ReactDOM.Render as third parameter should work, but empirically it doesn't. The timeout makes it reliable.
I could have queued this only for the auth case, but I think this way is a much better starting point, as this test will fail the moment you add a single http request to your app and it's not easy to figure out.